### PR TITLE
chore: Enable codeql action

### PR DESCRIPTION
**Pull request by naveensrinivasan**
This action runs GitHub's industry-leading semantic code analysis engine,
CodeQL, against a repository's source code to find security vulnerabilities.

https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql

https://github.com/ossf/scorecard/blob/main/docs/checks.md#sast
Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        # Override language selection by uncommenting this and choosing your languages
+        # with:
+        #   languages: go, javascript, csharp, python, cpp, java
+
+      # Autobuild attempts to build any compiled languages (C/C++, C#, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below).
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v2
```
Comment (everettraven) [inline .github/workflows/codeql.yml:29]: This should probably be modified to use the existing `make build` and/or `make install` targets as there are certain flags that get passed to the `go build` command that are likely to not be present in the autobuild step.

**Review by everettraven — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .github/workflows/codeql.yml]: I have taken care of this. PTAL. Thanks

**Comment by naveensrinivasan** [comment]
> I think this looks good, but I think it would be a good idea to have @jmrodri or @asmacdo to take a look at this as well before merging

A friendly ping

**Comment by jmrodri** [comment]
What is the plan for dealing with this action? Are we going to fail PRs because of them? Are we going to maintain these?

I like the idea but not sure if I want to maintain these long term. Do we have a report of the output to see how bad we are right now?

**Comment by everettraven** [comment]
> What is the plan for dealing with this action? Are we going to fail PRs because of them? Are we going to maintain these?
> 
> I like the idea but not sure if I want to maintain these long term. Do we have a report of the output to see how bad we are right now?

@jmrodri I think we can make it so that the check won't fail PRs. I'm not sure how much maintenance this action would need on our end. I personally think just having the reports to look at would be good, even if we don't fail PRs on it. I am working on generating a report on my fork to see how it says we are now.

**Comment by naveensrinivasan** [comment]
> > What is the plan for dealing with this action? Are we going to fail PRs because of them? Are we going to maintain these?
> > I like the idea but not sure if I want to maintain these long term. Do we have a report of the output to see how bad we are right now?
> 
> @jmrodri I think we can make it so that the check won't fail PRs. I'm not sure how much maintenance this action would need on our end. I personally think just having the reports to look at would be good, even if we don't fail PRs on it. I am working on generating a report on my fork to see how it says we are now.

👍

**Comment by everettraven** [comment]
Here is the current report that I was able to get from the CodeQL scan on my fork:
https://github.com/everettraven/operator-sdk/security/code-scanning

Looks pretty clean from what I can tell

**Pull request closed without merging**